frontend/src/Components/ShipsPlacement.js

```diff
@@ -0,0 +1,160 @@
+import BoardGame from './BoardGame';
+import Button from '@mui/material/Button';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import ToggleButton from '@mui/material/ToggleButton';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import { useState, useEffect } from "react";
+import Fab from '@mui/material/Fab';
+import { readyToPlay, deleteAShip, placeAShip } from '../Api';
+import { useNavigate } from 'react-router-dom';
+import styles from './Styles/ShipsPlacement.css';
+import Alert from '@mui/material/Alert';
+import Grid from '@mui/material/Grid';
+import CircularProgress from '@mui/material/CircularProgress';
+
+
+function ShipsPlacement(props) {
+    const navigate = useNavigate();
+    const [choosenShipInd, setChoosenShipInd] = useState('');
+    const [deleteShipFail, setDeleteShipFail] = useState(false);
+    const [placeShipFail, setPlaceShipFail] = useState('');
+    const [startGameFail, setStartGameFail] = useState(false);
+    const [shipRowCol, setShipRowCol] = useState({start: [], end: []});
+    const [isLoading, setIsLoading] = useState(false);
+    const alphabet = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z"];
```

Comment:
this is used in two different components, `ShipsPlacement` and `BoardGame`. Please move it to a separate file and import it from both of them.